src/js/components/Form/Form.js

```diff
@@ -94,6 +94,22 @@ const Form = forwardRef(
       return valid;
     };
 
+    // Remove any errors that we don't have any validations for anymore.
+    const filterErrorValidations = errors => {
+      const nextErrors = errors;
+      return Object.keys(nextErrors)
+        .filter(n => !validations.current[n] || nextErrors[n] === undefined)
+        .map(n => delete nextErrors[n]);
```

Comment:
I don't think we want `map` here, since nothing is depending on the return value. I suspect `forEach` would be better.

Reply:
done